spinnaker-jenkins-to-vmss/azuredeploy.json

```diff
@@ -0,0 +1,542 @@
+    "vnetName": "[concat(variables('scenarioPrefix'),'Vnet')]",
+    "vnetAddressSpace" : "10.0.0.0/20",
+    "subnet0Name": "[concat(variables('scenarioPrefix'),'Subnet0')]",
+    "subnet0AddressPrefix": "10.0.0.0/24",
+    "subnet1Name": "[concat(variables('scenarioPrefix'),'Subnet1')]",
+    "subnet1AddressPrefix": "10.0.1.0/24",
+    "subnet2Name": "[concat(variables('scenarioPrefix'),'Subnet2')]",
+    "subnet2AddressPrefix": "10.0.2.0/24",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))]",
+    "subnet0Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet0Name'))]",
+    "jenkinsNSGName": "jenkinsNSG",
+    "jenkinsNICName": "jenkinsNic",
+    "jenkinsPrivateIP": "10.0.0.5",
+    "jenkinsPublicIPAddressName": "jenkinsPublicIP",
+    "jenkinsVMName": "jenkinsVm",
+    "jenkinsOSDiskName": "jenkinsOSDisk",
+    "spinnakerNSGName" : "spinnakerNSG",
+    "spinnakerNICName" : "spinnakerNic",
+    "spinnakerPrivateIP" : "10.0.0.4",
+    "spinnakerPublicIPAddressName": "spinnakerPublicIP",
+    "spinnakerVMName": "spinnakerVm",
+    "spinnakerOSDiskName": "spinnakerOSDisk",
+    "keysPermissions": [ "all" ],
+    "secretsPermissions": [ "get" ]
+  },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[variables('front50StorageAccountName')]",
+            "apiVersion": "2016-01-01",
+            "location": "[resourceGroup().location]",
+            "kind": "Storage",
+            "sku": {
+                "name": "Standard_LRS"
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[variables('storageAccountName')]",
+            "apiVersion": "2016-01-01",
+            "location": "[resourceGroup().location]",
+            "kind": "Storage",
+            "sku": {
+                "name": "Standard_LRS"
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[variables('packerStorageAccountName')]",
+            "apiVersion": "2016-01-01",
+            "location": "[resourceGroup().location]",
+            "kind": "Storage",
+            "sku": {
+                "name": "Standard_LRS"
+            }
+        },
+        {
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2015-06-01",
+            "name": "[parameters('keyVaultAccountName')]",
+            "location": "[resourceGroup().location]",
+            "properties": {
+                "enabledForDeployment": true,
+                "enabledForTemplateDeployment": true,
+                "enabledForVolumeEncryption": true,
+                "tenantId": "[subscription().tenantId]",
+                "accessPolicies": [
+                    {
+                        "tenantId": "[subscription().tenantId]",
+                        "objectId": "[parameters('servicePrincipalClientId')]",
```

Comment:
From the keyvault documentation you can use the objectId or the serviceprincipal. The AppId is also present in the ServiceprincipalNames of the SPN object and therefore we are using it to reduce the number of parameters used.